res/controllers/Behringer-CMDStudio4a-scripts.js

```diff
@@ -0,0 +1,257 @@
+}
+
+BehringerCMDStudio4a.init = function () {
+    // Initialise anything that might not be in the correct state.
+    BehringerCMDStudio4a.initLEDs();
+    // Connect the VUMeters
+    engine.connectControl("[Master]","VuMeterL","BehringerCMDStudio4a.vuMeterUpdate");
+    engine.connectControl("[Master]","VuMeterR","BehringerCMDStudio4a.vuMeterUpdate");
+}
+ 
+BehringerCMDStudio4a.shutdown = function() {
+    // Leave the deck in a properly initialised state.
+    BehringerCMDStudio4a.initLEDs();
+
+    // Disconnect the VUMeters.
+    // Maybe not! This doesn't seem to work here!
+//  engine.connectControl("[Master]","VuMeterL","BehringerCMDStudio4a.vuMeterUpdate",true);
+//  engine.connectControl("[Master]","VuMeterR","BehringerCMDStudio4a.vuMeterUpdate",true);
+}
+
+
+// *************************** Control Stuff. *********************************
+// The code below is primarily "shift/mode" key functionality as there is no
+// native support for this in Mixxx at the moment.
+// I suspect that the vast majority of controller mappings could be completed
+// with little or no scripting if Mixxx supported shift/mode buttons in the
+// XML (together with standard wheel/scratching functionality).
+
+// Function to deal with the DEL "shift" buttons.
+BehringerCMDStudio4a.del = function (channel, control, value, status, group) {
+    BehringerCMDStudio4a.delState[channel]=!BehringerCMDStudio4a.delState[channel];
+    midi.sendShortMsg(status, control, BehringerCMDStudio4a.delState[channel] ? 0x01 : 0x00);
+}
+
+// Functions to deal with the play button.
+BehringerCMDStudio4a.playPush = function (channel, control, value, status, group) {
+    if (BehringerCMDStudio4a.delState[channel]) {
+        // DEL is active, reverse-roll (slip)!
+        engine.setValue(group, "reverseroll", 1);
+    } else {
+        // No DEL, just play.
+        if (engine.getValue(group, "play")==0) {
+            engine.setValue(group, "play", 1);
+        } else {
+            engine.setValue(group, "play", 0);
+        }
+    }
+}
+BehringerCMDStudio4a.playRelease = function (channel, control, value, status, group) {
+    if (BehringerCMDStudio4a.delState[channel]) {
+        // DEL is active, reverse-roll (slip) release!
+        engine.setValue(group, "reverseroll", 0);
+    }
+}
+
+// Functions to deal with the cue button.
+BehringerCMDStudio4a.cuePush = function (channel, control, value, status, group) {
+    if (BehringerCMDStudio4a.delState[channel]) {
+        // DEL is active, reverse play!
+        engine.setValue(group, "reverse", 1);
+    } else {
+        // No DEL so just cue.
+        engine.setValue(group, "cue_default", 1);
+    }
+}
+BehringerCMDStudio4a.cueRelease = function (channel, control, value, status, group) {
+    if (BehringerCMDStudio4a.delState[channel]) {
+        // DEL is active, reverse-roll release!
+        engine.setValue(group, "reverse", 0);
+    } else {
+        engine.setValue(group, "cue_default", 0);
+	}
+}

```

Comment:
I do think that is really unlikely and not really the job of a mapping to deal with. The most likely scenario I could think of in which this might possibly be an issue would be someone holding the cue button on one controller and accidentally hitting it on this controller before releasing it.